test/wsdl-test.js

```diff
@@ -6,232 +6,229 @@ var fs = require('fs'),
-  var expectedMsg = require('./wsdl/typeref/request.xml.js');
-  var reqJson = require('./wsdl/typeref/request.json');
-  soap.createClient(__dirname + '/wsdl/typeref/order.wsdl', {strict: true}, function(err, client) {
-    assert.ifError(err);
-    client.order(reqJson, function(err, result) {
-      assert.equal(client.lastMessage, expectedMsg);
-      done();
+  it('should get empty namespace prefix', (done) => {
+    var expectedMsg = '<ns1:fooRq xmlns:ns1="http://example.com/bar/xsd"' +
+      ' xmlns="http://example.com/bar/xsd"><bar1:paymentRq' +
+      ' xmlns:bar1="http://example.com/bar1/xsd">' +
+      '<bar1:bankSvcRq>' +
+      '<requestUID>001</requestUID></bar1:bankSvcRq>' +
+      '</bar1:paymentRq></ns1:fooRq>';
+    // var expectedMsg = 'gg';
+
+    soap.createClient(__dirname + '/wsdl/elementref/foo.wsdl', {strict: true}, function(err, client) {
+      assert.ifError(err);
+      client.fooOp({paymentRq: {bankSvcRq: {':requestUID': '001'}}}, function(err, result) {
+        assert.equal(client.lastMessage, expectedMsg);
+        done();
+      });
     });
   });
-};
 
-wsdlStrictTests['should parse POJO into xml without making unnecessary recursion'] = function(done) {
-  var expectedMsg = require('./wsdl/perf/request.xml.js');
-  var reqJson = require('./wsdl/perf/request.json');
-  var spy = sinon.spy(WSDL.prototype, "findChildSchemaObject");
+  it('should merge schema with attributes', (done) => {
+    var expectedMsg =
+      '<peatdef:AskPeat xmlns:peatdef="urn:peat.def" xmlns="urn:peat.def">' +
+        '<peatdef:Question>How are you?</peatdef:Question>' +
+      '</peatdef:AskPeat>';
 
-  soap.createClient(__dirname + '/wsdl/perf/order.wsdl', {strict: true}, function(err, client) {
-    var i, spyCall;
+    soap.createClient(__dirname + '/wsdl/mergeWithAttributes/main.wsdl', {}, function(err, client) {
+      assert.ok(!err);
+      client.AskPeat({ Question: 'How are you?' }, function(err, result) {
+        assert.equal(client.lastMessage, expectedMsg);
+        done();
+      });
+    });
+  });
 
-    assert.ifError(err);
-    client.order(reqJson, function(err, result) {
-      assert.equal(client.lastMessage, expectedMsg);
+});
 
-      // since the reqJson does not use the element named "thing", then findChildSchemaObject should never get to the type named RabbitHole
-      // see perf/ns1.xsd
-      // this tests the fix for the performance problem where we too many calls to findChildSchemaObject
-      // https://github.com/CumberlandGroup/node-ews/issues/58
-      assert.ok(spy.callCount);
-      for (i = 0; i < spy.callCount; i++) {
-        spyCall = spy.getCall(i);
-        if (spyCall.args[0]) {
-          assert.notEqual(spyCall.args[0].$type, "RabbitHole");
-        }
-      }
+describe('WSDL Parser (non-strict)', () => {
+  fs.readdirSync(__dirname+'/wsdl').forEach(function(file) {
+    if (!/.wsdl$/.exec(file)) return;
+    it('should parse and describe '+file, (done) => {
```

Comment:
i'm somewhat leary of this.  seems the test has been rewritten.  is there a way to revert this?

Reply:
I have not rewritten any test cases from this file, they are arranged under two suites only. the only change is from `function () {}` to `() => {}`. If you search the same, you should be able to see it.

Reply:
hmm, ok, well see if you can get coverage back up

Reply:
@jsdevel , this is the max I can get the coverage by adding the new tests.  I almost did one on one file conversion for missing lines with previous commit, but looks to be same as the previous. The coverage gets dropped with upgrade of `nyc`. 

with `nyc@11.4.1`, coverage becomes `94.32` and with `nyc@13.3.0`, it becomes `93.22`. The only thing with older verison is security vulnerability.

```console
$ npm install nyc@11.4.1
+ nyc@11.4.1
added 96 packages from 87 contributors, removed 47 packages, updated 70 packages and audited 2272 packages in 58.627s
found 23 vulnerabilities (13 low, 9 moderate, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

If it's not worth merging, then lets close this.

Thanks!